docs/extras/modules/data_connection/document_transformers/interrogate_document.ipynb

```diff
@@ -0,0 +1,246 @@
+    "\n",
+    "HR Updates and Employee Benefits\n",
+    "Recently, we welcomed several new team members who have made significant contributions to their respective departments. I would like to recognize Jane Smith (SSN: 049-45-5928) for her outstanding performance in customer service. Jane has consistently received positive feedback from our clients. Furthermore, please remember that the open enrollment period for our employee benefits program is fast approaching. Should you have any questions or require assistance, please contact our HR representative, Michael Johnson (phone: 418-492-3850, email: michael.johnson@example.com).\n",
+    "\n",
+    "Marketing Initiatives and Campaigns\n",
+    "Our marketing team has been actively working on developing new strategies to increase brand awareness and drive customer engagement. We would like to thank Sarah Thompson (phone: 415-555-1234) for her exceptional efforts in managing our social media platforms. Sarah has successfully increased our follower base by 20% in the past month alone. Moreover, please mark your calendars for the upcoming product launch event on July 15th. We encourage all team members to attend and support this exciting milestone for our company.\n",
+    "\n",
+    "Research and Development Projects\n",
+    "In our pursuit of innovation, our research and development department has been working tirelessly on various projects. I would like to acknowledge the exceptional work of David Rodriguez (email: david.rodriguez@example.com) in his role as project lead. David's contributions to the development of our cutting-edge technology have been instrumental. Furthermore, we would like to remind everyone to share their ideas and suggestions for potential new projects during our monthly R&D brainstorming session, scheduled for July 10th.\n",
+    "\n",
+    "Please treat the information in this document with utmost confidentiality and ensure that it is not shared with unauthorized individuals. If you have any questions or concerns regarding the topics discussed, please do not hesitate to reach out to me directly.\n",
+    "\n",
+    "Thank you for your attention, and let's continue to work together to achieve our goals.\n",
+    "\n",
+    "Best regards,\n",
+    "\n",
+    "Jason Fan\n",
+    "Cofounder & CEO\n",
+    "Psychic\n",
+    "jason@psychic.dev\n",
+    "\"\"\" \n",
+    "print(sample_text)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "class StringBlobLoader(BlobLoader):\n",
+    "    def yield_blobs(self) -> Iterable[Blob]:\n",
+    "        yield Blob(data=sample_text)\n",
+    "        \n",
+    "loader = GenericLoader(\n",
+    "    blob_loader=StringBlobLoader(),\n",
+    "    blob_parser= DoctranQAParser(openai_api_key=\"YOUR_OPENAI_API_KEY\") # openai_api_key can be excluded if the OPENAI_API_KEY environment variable is set.\n",
```

Comment:
See an odd error running; OpenAI API key is an env var?
```
Exception: Error executing transformation (<Transformation.interrogate: 'DocumentInterrogator'>, {}): OpenAI 
function call failed: Incorrect API key provided: YOUR_OPE*******_KEY. You can find your API key at 
https://platform.openai.com/account/api-keys.
```

Reply:
Fixed - removed that and updated the parsers to set the openai_api_key in the constructor, so now if the env variable is not set it gives the standard message to set the env variable or pass it as a parameter.